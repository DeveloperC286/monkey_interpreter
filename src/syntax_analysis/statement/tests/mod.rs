use rstest::rstest;

#[rstest(
    code,
    snapshot_name,
    case("return 5;", "test_syntax_analysis_for_return_statements_case1"),
    case(
        "return 5; return x;",
        "test_syntax_analysis_for_return_statements_case2"
    )
)]
fn test_syntax_analysis_for_return_statements(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name);
}

#[rstest(
    code,
    snapshot_name,
    case("let x = 5;", "test_syntax_analysis_for_let_statements_case1"),
    case(
        "let x = 5;let z = 7 + 10;",
        "test_syntax_analysis_for_let_statements_case2"
    )
)]
fn test_syntax_analysis_for_let_statements(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name);
}

#[rstest(
    code,
    snapshot_name,
    case("let x", "test_syntax_analysis_for_let_statements_syntax_errors_case1"),
    case("let", "test_syntax_analysis_for_let_statements_syntax_errors_case2")
)]
fn test_syntax_analysis_for_let_statements_syntax_errors(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name);
}

#[rstest(
    code,
    snapshot_name,
    case(
        "let x = 5; return 5;",
        "test_syntax_analysis_for_combined_statements_case1"
    )
)]
fn test_syntax_analysis_for_combined_statements(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name)
}
