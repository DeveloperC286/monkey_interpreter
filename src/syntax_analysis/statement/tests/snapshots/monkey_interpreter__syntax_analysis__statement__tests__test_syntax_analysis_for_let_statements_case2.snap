---
source: src/syntax_analysis/statement/tests/mod.rs
expression: "crate::syntax_analysis::SyntaxAnalysis::from(crate::lexical_analysis::LexicalAnalysis::from(code))"

---
AbstractSyntaxTree {
    abstract_syntax_tree: [
        Statement {
            statement: Let {
                identifier: "x",
                expression: Integer {
                    integer_token: Integer {
                        literal: "5",
                    },
                },
            },
        },
        Statement {
            statement: Let {
                identifier: "z",
                expression: Infix {
                    left_hand: Integer {
                        integer_token: Integer {
                            literal: "7",
                        },
                    },
                    operator_token: Plus,
                    right_hand: Integer {
                        integer_token: Integer {
                            literal: "10",
                        },
                    },
                },
            },
        },
    ],
    syntax_parsing_errors: [],
}
